Compartmentalize update scripts (for easier collaboration)

ridgekuhn opened the pull request:
_Note: This is not ready to merge, the script uses my fork's URLs for demo purposes_

I hope I'm not overstepping my bounds, but I've taken the liberty of streamlining the update script and breaking the updates into their own modules. This should make your life easier by elminating repetitive/error-prone tasks and preventing merge conflicts with collaborators. 

All you have to do to deploy a new update is make a directory with the required files and a script named `update.sh` to act on them, like this:

_/updates/RG351P/mynewupdate/update.sh:_
```shell
	#!/bin/bash
	# Arguments passed in by the main script
	UPDATE_NAME=${1} # 12202020
	UPDATE_DIR_URL=${2} # https://raw.githubusercontent.com/christianhaitian/arkos/main/updates/RG351P
	LOG_FILE=${3} # /home/ark/.update12202020.log

	# Do something...
	# curl ${UPDATE_DIR_URL}/${UPDATE_NAME}/update.sh | tee -a "${LOG_FILE}"
	exit
```

And that's it! The script handles update detection by parsing the GitHub repo, and takes care of naming, setting lock files, etc, based on the update directory path. Here's the [readme](https://github.com/ridgekuhn/arkos/tree/updaterefactor/updates). Also the [new update script](https://github.com/ridgekuhn/arkos/blob/updaterefactor/Update-RG351P.sh), and [/updates directory](https://github.com/ridgekuhn/arkos/tree/updaterefactor/updates). 

I've also done all the work of breaking the old update script up into the new modular format. A couple things I noticed:
* 11162020 already had an Update.sh which appears to be a copy of the script at /opt/system/Update.sh. I've renamed it to Update.sh.old since it doesn't seem to get called?
* The lock file for 11182020 was named 11182020-1, so the folders and scripts in the updates directory are configured appropriately for the new format.

I've tested by updating from 12162020 to 12192020 without issue, so if this looks good to you, all that needs to be done is delete the lines marked with `@todo remove this` and uncomment the ones below them!

ridgekuhn commented:
Rebased and updated to current version 12212020-1

christianhaitian commented:
Interesting.  How would the update option in ArkOS know that there is an additional update to be completed?

ridgekuhn commented:
@christianhaitian The script would [call the GitHub API](https://github.com/ridgekuhn/arkos/blob/f825852c6796312040f2b42d24680e60cc2218a3/Update-RG351P.sh#L64) and parse the main/default branch's [/updates/RG351P/](https://github.com/ridgekuhn/arkos/tree/updaterefactor/updates) directory to look for available updates (this is where I moved all the existing update directories to). So, all you have to do is create an update directory as you normally do, put a file named `update.sh` in it instead of adding on to the existing Update-RG351P.sh file, and push the commit. If the update directory is named 12222020 for example, the script will [automatically create](https://github.com/ridgekuhn/arkos/blob/f825852c6796312040f2b42d24680e60cc2218a3/Update-RG351P.sh#L131) the /home/ark/.update12222020 lock file like you've been using, would [handle errors](https://github.com/ridgekuhn/arkos/blob/f825852c6796312040f2b42d24680e60cc2218a3/Update-RG351P.sh#L126) in the /updates/RG351P/122220/update.sh file if it exits with code != 0, automatically [sets](https://github.com/ridgekuhn/arkos/blob/f825852c6796312040f2b42d24680e60cc2218a3/Update-RG351P.sh#L89)/[resets](https://github.com/ridgekuhn/arkos/blob/f825852c6796312040f2b42d24680e60cc2218a3/Update-RG351P.sh#L44) screen brightness, etc.

I realized while applying today's update, the one last redundancy I didn't account for is setting the version/date number, which we could automate by storing the files in updates/RG351P/1.5/12222020 and have the script also get the version number. If you're interested in this code branch, I'll code that up in a day or three

ridgekuhn commented:
@christianhaitian The main reason I wrote this is bc it avoids merge conflicts in Update-RG351P.sh, which takes the responsibility off your shoulders of having to integrate the update if other contributors are also writing them (and then they don't have to worry about resetting screen brightness, printing version numbers, etc). Instead, you would only have to approve the pull request

ridgekuhn commented:
I realized this PR's functionality would break after the new year due to the MMDDYYYY naming convention, so I've changed it to YYYYMMDD and added a line which looks for a lock file named either .updateMMDDYYYY-x or .updateYYYYMMD-x for backwards compatibility.
